packages/node-core/src/indexer/poi/poi.service.ts

```diff
@@ -1,55 +1,261 @@
+        (NOT EXISTS (SELECT 1 FROM ${tableName} WHERE "chainBlockHash" IS NOT NULL)) AS chainBlockHash_nullable,    
+        (NOT EXISTS (SELECT 1 FROM ${tableName} WHERE "hash" IS NOT NULL)) AS hash_nullable,
+        (NOT EXISTS (SELECT 1 FROM ${tableName} WHERE "parentHash" IS NOT NULL)) AS parent_nullable,
+        (EXISTS (SELECT 1 FROM information_schema.columns WHERE table_name = '_poi' AND column_name = 'mmrRoot' AND table_schema = '${schema}' )) AS mmr_exists;`;
+
+      const checkResult = await this.poiRepo.model.sequelize?.query(checkAttributesQuery, {
+        plain: true,
+      });
+
+      // Drop previous keys in metadata
+      this.storeCache.metadata.bulkRemove(['blockOffset', 'latestPoiWithMmr', 'lastPoiHeight']);
+
+      const queries = [];
+
+      if (checkResult) {
+        if (checkResult.mmr_exists) {
+          queries.push(`ALTER TABLE ${tableName} DROP COLUMN "mmrRoot";`);
+          queries.push(`DROP TABLE IF EXISTS "${schema}"."_mmr";`);
+        }
+        if (!checkResult.chainBlockHash_nullable) {
+          queries.push(`ALTER TABLE ${tableName} ALTER COLUMN "operationHashRoot" DROP NOT NULL;`);
+        }
+        if (!checkResult.chainBlockHash_nullable) {
+          queries.push(`ALTER TABLE ${tableName} ALTER COLUMN "chainBlockHash" DROP NOT NULL;`);
+          // keep existing chainBlockHash
+          queries.push(
+            `CREATE UNIQUE INDEX IF NOT EXISTS "poi_chainBlockHash" ON ${tableName} ("hash") WHERE "hash" IS NOT NULL`
+          );
+        }
+        if (!checkResult.hash_nullable) {
+          queries.push(`ALTER TABLE ${tableName} ALTER COLUMN "hash" DROP NOT NULL;`);
+          queries.push(`UPDATE ${tableName} SET hash = NULL;`);
+          queries.push(
+            `CREATE UNIQUE INDEX IF NOT EXISTS "poi_hash" ON ${tableName} ("hash") WHERE "hash" IS NOT NULL`
+          );
+        }
+        if (!checkResult.parent_nullable) {
+          queries.push(`ALTER TABLE ${tableName} ALTER COLUMN "parentHash" DROP NOT NULL;`);
+          queries.push(`UPDATE ${tableName} SET "parentHash" = NULL;`);
+          queries.push(
+            `CREATE UNIQUE INDEX IF NOT EXISTS "poi_parent_hash" ON ${tableName} ("parentHash") WHERE "parentHash" IS NOT NULL`
+          );
+        }
+      }
+
+      if (queries.length) {
+        for (const query of queries) {
+          try {
+            await this.poiRepo?.model.sequelize?.query(query, {type: QueryTypes.SELECT});
+          } catch (e) {
+            logger.error(`Migration poi failed with query: ${`query`}`);
+            throw e;
+          }
+        }
+        logger.info(`Successful migrate Poi`);
+      }
+      // Remove and Change column from sequelize not work, it only applies to public schema
+      // https://github.com/sequelize/sequelize/issues/13365
+      // await this.poiRepo?.model.sequelize?.getQueryInterface().changeColumn(tableName,'mmrRoot',{})
```

Comment:
Can we run raw sql?

Reply:
Current implementation is using raw sql, I left this comment here due to this bug here, we could not alter poi table columns with sequelize method `getQueryInterface().changeColumn`

Reply:
So the code above this is used instead of this?